src/v/cluster/archival/ntp_archiver_service.cc

```diff
@@ -1488,31 +1487,13 @@ ntp_archiver::maybe_upload_aborted_tx(
     co_return std::nullopt;
 }
 
-ss::future<> ntp_archiver::upload_index(
```

Comment:
Maybe not revert but just await the result of this method during the upload?
I'm not convinced that it should be reverted though.